ts/components/DevicePairingDialog.tsx

```diff
@@ -0,0 +1,293 @@
+import React from 'react';
+import { QRCode } from 'react-qrcode';
+
+import { SessionModal } from './session/SessionModal';
+import { SessionButton } from './session/SessionButton';
+
+interface Props {
+  i18n: any;
+  onClose: any;
+  pubKeyToUnpair: string | null;
+  pubKey: string | null;
+}
+
+interface State {
+  currentPubKey: string | null;
+  accepted: boolean;
+  isListening: boolean;
+  success: boolean;
+  loading: boolean;
+  view:
+    | 'default'
+    | 'waitingForRequest'
+    | 'requestReceived'
+    | 'requestAccepted'
+    | 'confirmUnpair';
+  pubKeyRequests: Array<any>;
+  data: Array<any>;
+}
+
+export class DevicePairingDialog extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props);
+
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.startReceivingRequests = this.startReceivingRequests.bind(this);
+    this.stopReceivingRequests = this.stopReceivingRequests.bind(this);
+    this.getPubkeyName = this.getPubkeyName.bind(this);
+
+    this.state = {
+      currentPubKey: this.props.pubKey,
+      accepted: false,
+      isListening: false,
+      success: false,
+      loading: true,
+      view: 'default',
+      pubKeyRequests: [],
+      data: [],
+    };
+  }
+
+  componentDidMount() {
+    this.getSecondaryDevices();
+  }
+
+  private async getSecondaryDevices() {
+    const secondaryDevices = await window.libloki.storage.getSecondaryDevicesFor(
+      this.state.currentPubKey
+    );
+    this.setState({
+      data: secondaryDevices,
+      loading: false,
+    });
+  }
+
+  public render() {
+    const { i18n } = this.props;
+
+    const newData = [
+      '053e18835c106a5f9f463a44a9d7ff9a26281d529285a047bd969cfc59d4ab8607',
+      '053e18835c106a5f9f463a44a9d7ff9a26281d529285a047bd969cfc59d4ab8604',
+    ];
+    setTimeout(() => {
+      this.setState({
+        data: newData,
+      });
+    }, 2000);
+
+    return (
+      <>
+        {!this.state.loading ? (
+          <SessionModal
+            title={i18n('pairedDevices')}
+            onOk={() => null}
+            onClose={this.closeDialog}
+          >
+            {this.state.view === 'waitingForRequest' ? (
```

Comment:
tslint does not like this kind of check in the rendering. You have to create a boolean at the start of the function and use it here